tests/src/3d/testqgs3drendering.cpp

```diff
@@ -964,5 +965,190 @@ void TestQgs3DRendering::testBillboardRendering()
   QVERIFY( renderCheck( "billboard_rendering_2", img2, 40 ) );
 }
 
+void TestQgs3DRendering::testColorRampShaderClassification()
+{
+  // Test Type::Interpolated and ClassificationMode::EqualInterval
+  {
+    std::unique_ptr<QgsColorRampShader> shader( new QgsColorRampShader( 0.0, 255.0, new QgsGradientColorRamp( Qt::green, Qt::blue ), QgsColorRampShader::Type::Interpolated, QgsColorRampShader::ClassificationMode::EqualInterval ) );
+    shader->classifyColorRamp( 5, -1 );
+
+    QList<QgsColorRampShader::ColorRampItem> itemsList = shader->colorRampItemList();
+    QList<QgsColorRampShader::ColorRampItem> itemsList2;
+    itemsList2.append( QgsColorRampShader::ColorRampItem( 0, QColor( 0,  255,  0 ),  "0" ) );
+    itemsList2.append( QgsColorRampShader::ColorRampItem( 63.75, QColor( 0,  191,  64 ),  "63.8" ) );
+    itemsList2.append( QgsColorRampShader::ColorRampItem( 127.5, QColor( 0,  128,  128 ),  "128" ) );
+    itemsList2.append( QgsColorRampShader::ColorRampItem( 191.25, QColor( 0,  64,  191 ),  "191" ) );
+    itemsList2.append( QgsColorRampShader::ColorRampItem( 255, QColor( 0,  0,  255 ),  "255" ) );
+
+    QVERIFY( itemsList.size() == itemsList2.size() );
+    for ( int i = 0; i < itemsList.size(); ++i )
+    {
+      QVERIFY( itemsList[i].value == itemsList2[i].value );
+      QVERIFY( itemsList[i].color.red() == itemsList2[i].color.red() );
+      QVERIFY( itemsList[i].color.green() == itemsList2[i].color.green() );
+      QVERIFY( itemsList[i].color.blue() == itemsList2[i].color.blue() );
+      QVERIFY( itemsList[i].color.alpha() == itemsList2[i].color.alpha() );
+    }
```

Comment:
this code for verification is repeated many times below... I would suggest to put them into some function - or even better, replace those with `QCOMPARE( itemsList, itemsList2 );` ... that should use `operator==` or `operator!=` of individual items to decide whether they are equal or not.

Reply:
The problem is that for some reason the equality test between 2 QColor instances is returning false even though they are printed as equal :(